docs/guides/prefect-deploy.md

```diff
@@ -56,593 +80,995 @@ flowchart LR
+
+Use the tabs below to explore both deployment creation options.
+
+=== ".deploy"
+
+    ### Automatically bake your code into a Docker image 
+
+    You can create a deployment from Python code by calling the `.deploy` method on a flow.
+
+    ```python hl_lines="17-22" title="buy.py"
+    from prefect import flow
+
+
+    @flow(log_prints=True)
+    def buy():
+        print("Buying securities")
+
+
+    if __name__ == "__main__":
+        buy.deploy(
+            name="my-code-baked-into-an-image-deployment", 
+            work_pool_name="my-docker-pool", 
+            image="my_registry/my_image:my_image_tag"
+        )
+    ```
+
+    Make sure you have the [work pool](/concepts/work-pools/) created in the Prefect Cloud workspace your are authenticated to or on your running self-hosted server instance connected to.  
+    Then run the script to create a deployment (in future examples this step will be omitted for brevity):
+
+    <div class="terminal">
+    ```bash
+    python buy.py
+    ```
+    </div>
+
+    You should see messages in your terminal that Docker is building your image.
+    When the deployment build succeeds you will see helpful information in your terminal showing you how to start a worker for your deployment and how to run your deployment.
+    Your deployment will be visible on the `Deployments` page in the UI.
+
+    By default, `.deploy` will build a Docker image with your flow code baked into it and push the image to the [Docker Hub](https://hub.docker.com/) registry specified in the `image` argument`. 
+    
+    !!! note "Authentication to Docker Hub"
+        You need your environment to be authenticated to your Docker registry to push an image to it.
+
+    You can specify a registry other than Docker Hub by providing the full registry path in the `image` argument.
+
+    !!! warning
+        If building a Docker image, the environment in which you are creating the deployment needs to have Docker installed and running.
+        The [`prefect-docker` package](https://prefecthq.github.io/prefect-docker/) also needs to be installed.
+
+    To avoid pushing to a registry, set `push=False` in the `.deploy` method.
+    
+    ```python hl_lines="6"
+
+    if __name__ == "__main__":
+        buy.deploy(
+            name="my-code-baked-into-an-image-deployment", 
+            work_pool_name="my-docker-pool", 
+            image="my_registry/my_image:my_image_tag",
+            push=False
+        )
+    ```
+
+    To avoid building an image, set `build=False` in the `.deploy` method.
+
+    ```python hl_lines="6"
+
+    if __name__ == "__main__":
+        buy.deploy(
+            name="my-code-baked-into-an-image-deployment", 
+            work_pool_name="my-docker-pool", 
+            image="discdiver/no-build-image:1.0",
+            build=False
+        )
+    ```
+
+    At runtime, the specified image will need to be available to access your flow code.
```

Comment:
```suggestion
    The specified image will need to be available in your deployment's execution environment for your flow code to be accessible.
```
Trying to differentiate that "runtime" is where the flow runs, not where you're running `.deploy`.